src/types.ts

```diff
@@ -42,11 +44,21 @@ export type UploadcareImageDetails = {
 type ImageSourceProps =
     | {
           src: string;
+          /**
+           * This prop only works in conjunction with `secondaryImageBreakpoints` prop. See its description for details.
+           */
+          secondarySrc?: string;
           imageDetails?: never;
+          secondaryImageDetails?: never;
       }
     | {
           src?: never;
+          secondarySrc?: never;
           imageDetails: UploadcareImageDetails;
+          /**
+           * This prop only works in conjunction with `secondaryImageBreakpoints` prop. See its description for details.
+           */
+          secondaryImageDetails?: UploadcareImageDetails;
```

Comment:
I have a feeling that the external API for this component looks too complex.

Has it ever been considered to mimic the standard [`picture` HTML element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/picture) behavior?

Something like this:

```tsx
<Picture fallback={fallbackUploadcareImage}>
   <Image breakpoints="..." sizes="..." src={uploadcareImage} />
   <Image breakpoints="..." sizes="..." src={secondaryUploadcareImage} />
</Image>
```

Wouldn't it be a simpler API to work with?

Reply:
Hmm, that looks much cleaner!
But I think we need to retain compatibility for single `Image` components, cause it wouldn't make sense for those to be wrapped in `Picture` component. But in this case we would have two new components 🤔 

This whole solution was created on the fly yesterday, as an answer to the problem that emerged the day before when creating this new hero block for the website, so it definitely could be made more nice, but I also would like to push the fix sooner than later, cause every page visitor loads an extra 1-1.5 Mb image for each slide on the homepage 😕 

This would be nice to figure out on a pair coding session, if you have time and energy to do it :)

Reply:
I published a preview version with the current implementation, since I won't have time work on this refactor in the next few days. I'll log a ticket to refactor the API later on.